---
title: static-jsx
description: The <T> component must only have static children
---

## Overview

Ensures that dynamic content inside [`<T>`](/docs/react/api/components/t) components is wrapped in variable components.

This rule flags dynamic content that appears directly inside [`<T>`](/docs/react/api/components/t) components without being wrapped in appropriate variable components like [`<Var>`](/docs/react/api/components/var).

## Reference

### Options

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `libs` | `string[]` | `["gt-react", "gt-next", "gt-react-native", "gt-i18n", "@generaltranslation/react-core"]` | Array of library modules to check for translation components |

---

## Examples

### ❌ Incorrect

Dynamic content appears directly in the `<T>` component:

```jsx
<T>Hello {userName}!</T>
```

### ✅ Correct

Dynamic content is wrapped in a variable component:

```jsx
<T>Hello <Var>{userName}</Var>!</T>
```

---

## Configuration

```json
{
  "@generaltranslation/react-core-linter/static-jsx": ["error", {
    "libs": ["gt-react", "gt-next", "gt-react-native"]
  }]
}
```